src/AutoRest.CSharp/Common/AutoRest/Communication/StandaloneGeneratorRunner.cs

```diff
@@ -62,7 +62,7 @@ public static async Task RunAsync(CommandLineOptions options)
                 workspace = await new CSharpGen().ExecuteAsync(codeModel);
                 if (options.IsNewProject)
                 {
-                    new CSharpProj().Execute(Configuration.Namespace, outputPath, (yaml.Contains("x-ms-format: dfe-", StringComparison.Ordinal)));
+                    new CSharpProj().Execute(Configuration.Namespace, outputPath, (yaml.Contains("x-ms-format: dfe-", StringComparison.Ordinal)), Configuration.ToCSharpProjConfiguration());
```

Comment:
I assume since part of your PR is making the test cases running sequentially, the concurrent issue is resolved.
Therefore the changes related with the new `CSharpProjConfiguration` is purely refactor and not resolving any issue, is this correct?
The idea of making the configuration class static is "to avoid the fact that we have to passing an instance around everywhere", which is exactly the thing we are doing right here. See the PR here: https://github.com/Azure/autorest.csharp/pull/2005
What is the advantage of having this new record of configuration?
Or if we have to have it, we could let the `Configuration` static class to hold it internally and redirect its values in static properties.

Reply:
> Therefore the changes related with the new CSharpProjConfiguration is purely refactor and not resolving any issue, is this correct?

Not correct. See the PR title. It is resolving the problem when two threads initialize configuration together. So I'm afraid your below solution is not going to work.

Reply:
In the actual executions, our generator never runs in parallel (they should be running in different process). And we are making the tests to run in sequence. Therefore the issue should be gone.